app/src/main/java/com/tz/sanga/moviestore/Fragments/First/FirstPresenter.java

```diff
@@ -0,0 +1,78 @@
+package com.tz.sanga.moviestore.Fragments.First;
+
+
+import com.tz.sanga.moviestore.BuildConfig;
+import com.tz.sanga.moviestore.Model.API.Connector;
+import com.tz.sanga.moviestore.Model.API.Service;
+import com.tz.sanga.moviestore.Model.Movie;
+import com.tz.sanga.moviestore.Model.MoviesResponse;
+import com.tz.sanga.moviestore.Model.Trailer;
+import com.tz.sanga.moviestore.Model.TrailerResponse;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class FirstPresenter {
+    private static final int PAGE_START = 1;
+    private int currentPage = PAGE_START;
+    private FirstView firstView;
+    private Service service;
+    private int moveId;
+
+    public FirstPresenter(FirstView firstView, int moveId) {
+        this.firstView = firstView;
+        this.moveId = moveId;
+    }
+
+    public void updateMoveId(int moveId){
+        this.moveId = moveId;
+    }
+
+    public void getData(){
+        firstView.showLoading();
+        service = Connector.getConnector().create(Service.class);
```

Comment:
This could only be initialized once in the constructor of the firstPresenter the use it all over the class